src/rtextures.c

```diff
@@ -291,11 +291,21 @@ Image LoadImage(const char *fileName)
     // Loading file to memory
     int dataSize = 0;
     unsigned char *fileData = LoadFileData(fileName, &dataSize);
+    const char *fileType = GetFileExtension(fileName);
+
+    #if defined(SUPPORT_FILEFORMAT_SVG)
```

Comment:
I prefer to avoid this hack to consider a specific format situation, it breaks the funtion symmetry and organization.

Reply:
no problem, I can see a few options to this.

1. I'm not sure if this would break symmetry as well, but moving all of SVG functionality  from `LoadImage` `LoadImageFromMemory` to `LoadImageSvg` would be very simple

2. nanosvg.h could be altered, so it's functions take a file_data and length parameter instead of relying on the data string being formatted.
This gets rid of the `realloc()` calls, and makes the functions safer in my opinion.

3. `LoadImage` can be made symmetric again by pushing the buffer re allocation call to `LoadImageFromMemory`.
Originally I avoided doing this, as a non-allocated string could be passed to `LoadImageFromMemory`, and trying to reallocate that would exit the program very 'inelegantly',
But an argument can be made that the chance of a non-allocated string somehow being not NULL terminated is very low

Reply:
Very good alternatives list! Despite I don't like changing external libraries I think the best option for raylib is 2.

My other alternative to avoid this hack is just removing SVG support, users with this requirement could just implement it themselfs, actually an example could be added in `raylib-extras`.

`LoadImageSvg()` is really and UGLY function and afaik, the only function in all the library file-format specific, that is something I tried to avoid in raylib...

Definitely, moving SVG support to an example seems the best solution to me...

Reply:
On trying to apply option 2, a small problem appeared

The issue is fixed with `LoadImage()`, `LoadImageFromMemory()`,
and `LoadImageSvg()` behaves nicely when loading from files

But `LoadImageSvg()` from a string still has the issue that the input needs to be a NULL terminated string. Since the function has no parameter for the size of the string

The tweak does 'shrink' the issue down to only one use case though:
`LoadImageSvg(LoadFileData())`